repositories/employeeRepository.go

```diff
@@ -16,71 +16,317 @@ type EmployeeRepository struct {
+		err := employee.Id.AssignTo(&employeeUUID)
 		if err != nil {
-			return nil, err
+			return allEmployees, err
 		}
-		allEmployees = append(allEmployees, employee)
+
+		var companyUUID string
+		err = employee.Idc.AssignTo(&companyUUID)
+		if err != nil {
+			return allEmployees, err
+		}
+
+		allEmployees = append(allEmployees, models.Employee{
+			ID:        employeeUUID,
+			FirstName: employee.Firstname,
+			LastName:  employee.Lastname,
+			CompanyID: companyUUID,
+			C:         employee.C,
+			R:         employee.R,
+			U:         employee.U,
+			D:         employee.D,
+		})
 	}
 	return allEmployees, nil
 }
 
 // GetEmployeeByID .
 func (repository *EmployeeRepository) GetEmployeeByID(id string) (models.Employee, error) {
 	var employee models.Employee
-	err := repository.DB.QueryRow(context.Background(), "select * from employees where id=$1", id).Scan(&employee.ID, &employee.FirstName, &employee.LastName, &employee.CompanyID, &employee.C, &employee.R, &employee.U, &employee.D)
+
+	Uuid, err := uuid.FromString(id)
+	if err != nil {
+		return employee, err
+	}
+
+	rows, err := repository.DB.Query(context.Background(), "select * from employees where id=$1", Uuid)
+	defer rows.Close()
+
 	if err != nil {
 		return employee, err
 	}
+
+	if !rows.Next() {
+		return employee, errors.New("There is no employee with this id")
+	}
+
+	var employeePers persistence.Employees
+	employeePers.Scan(&rows)
+
+	var employeeUUID string
+	err = employeePers.Id.AssignTo(&employeeUUID)
+	if err != nil {
+		return employee, err
+	}
+
+	var companyUUID string
+	err = employeePers.Idc.AssignTo(&companyUUID)
+	if err != nil {
+		return employee, err
+	}
+
+	employee = models.Employee{
+		ID:        employeeUUID,
+		FirstName: employeePers.Firstname,
+		LastName:  employeePers.Lastname,
+		CompanyID: companyUUID,
+		C:         employeePers.C,
+		R:         employeePers.R,
+		U:         employeePers.U,
+		D:         employeePers.D,
+	}
+
 	return employee, nil
 }
 
 // AddEmployee .
 func (repository *EmployeeRepository) AddEmployee(employee *models.Employee) error {
-	u := uuid.NewV4()
-	employee.ID = u.String()
-	_, err := repository.DB.Exec(context.Background(), "insert into employees (id, firstname, lastname, idc, c, r, u, d) values ($1, $2, $3, $4, $5, $6, $7, $8)",
-		employee.ID, employee.FirstName, employee.LastName, employee.CompanyID, employee.C, employee.R, employee.U, employee.D)
+	tx, err := repository.DB.Begin(context.Background())
 	if err != nil {
 		return err
 	}
+	defer tx.Rollback(context.Background())
+
+	employee.ID = uuid.NewV4().String()
+
+	employeePers := persistence.Employees{
+		Firstname: employee.FirstName,
+		Lastname:  employee.LastName,
+		C:         employee.C,
+		R:         employee.R,
+		U:         employee.U,
+		D:         employee.D,
+	}
+	employeePers.Idc.Set(employee.CompanyID)
+	employeePers.Id.Set(employee.ID)
+
+	_, err = employeePers.InsertTx(&tx)
+	if err != nil {
+		return err
+	}
+
+	tx.Commit(context.Background())
```

Comment:
You could return this instead of nil below, Commit returns err, which would indicate if the transaction was a success.